File: electrowheelweb/src/services/AdminService.js
import axios from 'axios'

const ADMIN_LINK = `http://localhost:8080/admins`;

class AdminService{

     getAllAdmins(){
    return axios.get(ADMIN_LINK);
    }

    deleteAdmin(admin){
        return axios.delete(ADMIN_LINK, admin);
    }

    insertAdmin(admin){
        return axios.post(ADMIN_LINK, admin);
    }

       updateAdmin(admin){
        return axios.put(ADMIN_LINK, admin);
    }

    adminLogin(email, password){
        console.log(password);
        return axios.get(`http://localhost:8080/admins/login/${email}-${password}`);
    }


    getAdminById(id){
        return axios.get(ADMIN_LINK + `/${id}`);
    }

}

export default new AdminService()

File: ElectroWheel/src/main/java/com/ElectroWheelBackEnd/Service/AdminService.java
package com.ElectroWheelBackEnd.Service;

import com.ElectroWheelBackEnd.Dao.AdminDao;
import com.ElectroWheelBackEnd.Models.Admin;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service

public class AdminService {

    private static final Logger LOGGER = LoggerFactory.getLogger(AdminService.class);

    private final AdminDao adminLogInDao;
    @Autowired
    public AdminService(AdminDao adminLogInDao) {
        this.adminLogInDao = adminLogInDao;
    }

    public void createAdmin(Admin admin){
        adminLogInDao.insertAdmin(admin);
    }

    public void deleteAdmin(Admin admin){
        adminLogInDao.deleteAdmin(admin);
    }

    public void updateAdmin(Admin admin){
        adminLogInDao.updateAdmin(admin.getIdAdmin(), admin.getEmail(), admin.getPassword());
    }

    public Admin retrieveAdminById(int id){
        return adminLogInDao.retrieveAdminById(id);
    }

    public Boolean adminLogIn(String email, String password){
        return adminLogInDao.adminLogIn(email, password);
    }


    public List<Admin> retrieveAdmins() {
        LOGGER.info("Accessing service for admin retrieval");
        return adminLogInDao.retrieveAdmin();
    }

}
